projects/storefrontapp/project.json

```
@@ -141,36 +157,57 @@
           "namedChunks": false,
           "aot": true,
           "extractLicenses": true,
-          "vendorChunk": false,
-          "buildOptimizer": true,
           "serviceWorker": false,
           "tsConfig": "projects/storefrontapp/tsconfig.app.prod.json"
         },
+        "noSsr": {
+          "ssr": false,
+          "prerender": false
+        },
         "development": {
-          "buildOptimizer": false,
           "optimization": false,
-          "vendorChunk": true,
           "extractLicenses": false,
-          "sourceMap": true,
-          "namedChunks": true
+          "sourceMap": true
+        },
+        "prerender": {
+          "optimization": true,
+          "outputHashing": "all",
+          "sourceMap": false,
+          "namedChunks": false,
+          "serviceWorker": false,
+          "server": "projects/storefrontapp/src/main.server.ts",
+          "prerender": {
+            "discoverRoutes": false,
+            "routesFile": "projects/storefrontapp/prerender.txt"
+          }
```

[MINOR]
If possible, let's not duplicate here config properties specific for "production" or "development" build target (e.g. let's not include `optimization: true`, etc.) in the "prerender" section. Let's leave here only things needed for the prerendering - which is just  one sub-property: "prerender".

And then let's update other "scripts" in `package.json`, to include 2 suffixes: both the `:prerender` and the `:development`/`:production` , e.g. `storefrontapp:build:production:prerender`

In other words, I suggest to leave here only:
```suggestion
        "prerender": {
          "prerender": {
            "discoverRoutes": false,
            "routesFile": "projects/storefrontapp/prerender.txt"
          }
```

---

In NX, configurations cannot be merged (https://github.com/nrwl/nx/discussions/22710). That is why there are some duplicated values here for prerender.

---

Thank you for linking the discussion in the Nx repo. I've found there a workaround that allows merging configurations and avoid duplication of properties among configurations - proposed by the Nx maintainer.

I've prepared a proposal sub-PR for this PR with this workaround implementation: 
https://github.com/SAP/spartacus/pull/20355 . See that sub-PR description for more details.

I'm looking forward to hearing what you think about it.
If by any chance you like it, feel free to merge. If not, no worries, just let me know.